packages/cyberstorm/src/components/DropDown.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { ReactElement, ReactNode, useState } from "react";
+import styles from "./componentStyles/DropDown.module.css";
+import { Button } from "./Button";
+import { faChevronDown } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import * as DropdownMenu from "@radix-ui/react-dropdown-menu";
+
+type DropDownProps = {
+  label?: string;
+  icon?: ReactNode;
+  defaultOpen: boolean;
+  dropDownData?: DataItem[];
+  colorScheme?: "default" | "defaultDark" | "primary";
+  defaultValue?: number;
+};
+
+type DataItem = {
+  label: string;
+  value: number;
+  reactElement?: ReactElement;
+};
+
+export const DropDown: React.FC<DropDownProps> = (props) => {
+  const { icon, colorScheme, defaultOpen, dropDownData, defaultValue } = props;
+
+  let initialValue = dropDownData ? dropDownData[0] : null;
+  if (defaultValue && dropDownData && dropDownData[defaultValue]) {
+    initialValue = dropDownData[defaultValue];
+  }
+  const [value, setValue] = useState(initialValue);
+  const dropDownItemElements = mapDropDownData(
+    dropDownData,
+    colorScheme,
+    setValue
+  );
+
+  return (
+    <div className={styles.root}>
+      <DropdownMenu.Root defaultOpen={defaultOpen}>
```

Comment:
Should we check that if `dropDownData` is `undefined` or an empty array, the dropdown can't be opened (by `defaultOpen` or by the user manually)?

Reply:
@gkurck should this make the DropDown be disabled or something?